DNN Platform/Library/Entities/Portals/PortalController.cs

```diff
@@ -414,21 +426,13 @@ public static string GetEncryptedString(string settingName, int portalID, string
         /// <param name="portalID">The portal ID.</param>
         /// <param name="defaultValue">The default value.</param>
         /// <returns>Returns setting's value if portal contains the specific setting, otherwise return defaultValue.</returns>
+        [Obsolete("Deprecated in 9.8.0 Scheduled for removal in v11.0.0, use DotNetNuke.Abstractions.Portals.IPortalSettingsManager instead.")]
         public static string GetPortalSetting(string settingName, int portalID, string defaultValue)
         {
-            var retValue = Null.NullString;
-            try
-            {
-                string setting;
-                Instance.GetPortalSettings(portalID).TryGetValue(settingName, out setting);
-                retValue = string.IsNullOrEmpty(setting) ? defaultValue : setting;
-            }
-            catch (Exception exc)
-            {
-                Logger.Error(exc);
-            }
-
-            return retValue;
+            var portalSettingsManager = Globals.DependencyProvider.GetRequiredService<IPortalSettingsManager>();
+            return portalSettingsManager
+                .GetPortalSettings(portalID)
+                .Get<string>(settingName) ?? defaultValue;

```

Comment:
The previous logic is to return the `defaultValue` when the value is an empty string, is this an intentional change to only return `defaultValue` when the value is `null`?